files/statvfs_facts.py

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/python
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: statvfs
+version_added: "2.0"
+short_description: retrieve file system consumption
+description:
+     - Retrieves facts for a filesystem using os.statvfs.
+options:
+  path:
+    description:
+      - The full path of the filesystem to get the facts of
+    required: true
+    default: null
+    aliases: []
+  unit:
+    description:
+      - Convert size report to this unit (K, M, G)
+    required: false
+    default: o
+    aliases: []
+author: Yannig Perre
+'''
+
+EXAMPLES = '''
+# Obtain the stats of /tmp, and check space available
+- statvfs_facts: path=/tmp
+- fail: msg="Whoops! /tmp does not have enough space"
+  when: statvfs.space_available < 10000000
+'''
+
+RETURN = '''
+statvfs:
+    description: dictionary containing all the statvfs data
+    returned: success
+    type: dictionary
+    contains:
+'''
+
+import os
+import sys
+import pwd
+import grp
+
+def main():
+    # Size convertion
+    size_convertion = {
```

Comment:
module.pretty_bytes function might work here, no need to specify size, the most appropriate will be used (sorry I missed this the first time).